plugin/src/main/kotlin/com/vanniktech/maven/publish/sonatype/SonatypeRepositoryBuildService.kt

```diff
@@ -163,14 +219,75 @@ internal abstract class SonatypeRepositoryBuildService :
         if (buildIsSuccess) {
           throw e
         } else {
-          logger.info("Failed processing $stagingRepositoryId staging repository after previous build failure", e)
+          logger.info("Failed processing $uploadId staging repository after previous build failure", e)
         }
       }
     }
   }
 
   private fun runEndOfBuildActions(actions: List<EndOfBuildAction>) {
-    var stagingRepositoryId = stagingRepositoryId
+    if (parameters.sonatypeHost.get().centralPortal) {
+      runCentralPortalEndOfBuildActions(actions)
+    } else {
+      runNexusEndOfBuildActions(actions)
+    }
+  }
+
+  private fun runCentralPortalEndOfBuildActions(actions: List<EndOfBuildAction>) {
+    val uploadId = uploadId
+
+    val closeActions = actions.filterIsInstance<EndOfBuildAction.Close>()
+    if (closeActions.isNotEmpty()) {
+      if (uploadId != null) {
+        val deploymentName = "${parameters.groupId.get()}-$uploadId"
+        val publishingType = if (actions.contains(EndOfBuildAction.ReleaseAfterClose)) {
+          "AUTOMATIC"
+        } else {
+          "USER_MANAGED"
+        }
+
+        val directory = File(publishingUrl(false).substringAfter("://"))
```

Comment:
Won't this fail on Windows?

Reply:
You mean because of the `substringAfter("://")`? I think it should be fine since it's just removing the prefix that we add ourselves inside `publishingUrl`